olive/cli/quantize.py

```diff
@@ -0,0 +1,148 @@
+                "load_dataset_config": {},
+                "pre_process_data_config": {},
+                "dataloader_config": {},
+                "post_process_data_config": {},
+            }
+        ],
+        "passes": {
+            "awq": {"type": "AutoAWQQuantizer"},
+            "gptq": {
+                # Ref: https://github.com/AutoGPTQ/AutoGPTQ/pull/651/files
+                "type": "GptqQuantizer",
+                "data_config": "default_data_config",
+            },
+            "quarot": {
+                "type": "QuaRot",
+                "w_rtn": True,
+                "rotate": True,
+                "w_bits": 4,
+                "a_bits": 4,
+                "k_bits": 4,
+                "v_bits": 4,
+                "calibration_data_config": None,
+            },
+        },
+        "pass_flows": [],
+        "output_dir": "models",
+        "host": "local_system",
+        "target": "local_system",
+    }
+
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        sub_parser = parser.add_parser(
+            "quantize",
+            help="Quantize the input model",
+        )
+
+        # model options
+        add_input_model_options(sub_parser, enable_hf=True, enable_pt=True, default_output_path="quantized-model")
+
+        # Logging options
+        add_logging_options(sub_parser)
+
+        sub_parser.add_argument(
+            "--algorithms",
+            type=str,
+            nargs="*",
+            required=True,
+            choices=sorted(QuantizeCommand._CONFIG_TEMPLATE["passes"].keys()),
+            help="List of quantization algorithms to run.",
+        )
+
+        # dataset options
+        add_dataset_options(sub_parser)
+
+        # accelerator options
+        add_accelerator_options(sub_parser)
+
+        # remote options
+        add_remote_options(sub_parser)
+
+        sub_parser.set_defaults(func=QuantizeCommand)
+
+    def _get_run_config(self, tempdir: str) -> Dict[str, Any]:
+        config = deepcopy(QuantizeCommand._CONFIG_TEMPLATE)
+        update_input_model_options(self.args, config)
+        update_dataset_options(self.args, config)
+        update_accelerator_options(self.args, config)
+        config["log_severity_level"] = self.args.log_level
+
+        to_replace = [
+            (("pass_flows"), [[name] for name in self.args.algorithms]),
+            (("output_dir"), tempdir),
+        ]
+        for k, v in to_replace:
+            if v is not None:
+                set_nested_dict_value(config, k, v)
+
+        return config
+
+    def run(self):
+        from olive.workflows import run as olive_run
+
+        has_failures = False
+        with tempfile.TemporaryDirectory(prefix="olive-cli-tmp-", dir=self.args.output_path) as tempdir:
+            run_config = self._get_run_config(tempdir)
+            olive_run(run_config)
+
+            if is_remote_run(self.args):
+                return
+
+            output_path = Path(self.args.output_path)
```

Comment:
https://github.com/microsoft/Olive/blob/180dffb8af7643d4dfcffcb044a14ba171ab2647/olive/cli/base.py#L410 handles this automatically. If there is only one pass_flow, the `algorithm_name` nesting doesn't exist.

Reply:
The pass_flows never have more than one entry. The nesting does exist in either case. It is verified as part of unit test as well - parameterized with one pass and two pass_flows.

Reply:
Nevermind, I think you mean in the output folder of the cli, not in the output folder of the olive_run.